### PR TITLE
Reordering recording, sorting args

### LINKED ISSUE #3086: Standardise order of recording, sorting args
Hi all! I've noticed that for different functions which require both a recording and a sorting, the order in which these should be passed is not consistent. E.g. `si.create_sorting_analyzer(sorting, recording)` but `si.estimate_sparsity(recording, sorting)`

Not exactly high priority, but it might be nice to standardise this order in the API at some point? Although I can also see how it might be better to have the first argument be the most relevant object for whatever that function is doing

### PR DESCRIPTION
**Pull request by jakeswann1**
Would fix #3086, if we want to break `estimate_sparsity`

### CODE REVIEW

**Review by zm711 — COMMENTED** [review]
this looks good to me. I think @alejoe91 and @samuelgarcia will decide on whether they are okay with the break. But I think it is the right idea.

**Comment by alejoe91** [comment]
This is ok with me! Thanks @jakeswann1 

@samuelgarcia ?

**Comment by samuelgarcia** [comment]
This is lotally OK for me.
I hope to not regret that we merge 2 weeks before the relase.
Lets see if any bombs blow up after push the merge button.

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**
